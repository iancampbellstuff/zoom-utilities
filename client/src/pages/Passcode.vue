<template>
    <q-page id="passcode" class="row items-start justify-evenly">
        <q-form
            id="form"
            class="q-gutter-lg"
            @change="onChange"
            @submit="onSubmit"
            @reset="onReset"
            greedy
        >
            <q-toolbar-title class="text-center"
                >Generate a Passcode</q-toolbar-title
            >
            <q-input
                type="number"
                :min="MINIMUM_PASSCODE_LENGTH"
                :max="MAXIMUM_PASSCODE_LENGTH"
                v-model="length"
                label="Length"
                hint="(optional)"
                outlined
                standout
                dense
            />
            <q-input
                type="number"
                :min="MINIMUM_DIGIT"
                :max="MAXIMUM_DIGIT"
                v-model="firstDigit"
                label="First Digit"
                hint="(optional)"
                outlined
                standout
                dense
            />
            <q-btn-group spread>
                <q-btn label="Submit" type="submit" color="primary" no-caps />
                <q-btn
                    label="Reset"
                    type="reset"
                    color="primary"
                    class="q-ml-sm"
                    no-caps
                    flat
                />
            </q-btn-group>
            <q-btn-group spread>
                <q-input
                    v-model="passcode"
                    label="Passcode"
                    outlined
                    standout
                    dense
                    filled
                    readonly
                />
                <q-btn color="secondary" @click="onCopy" no-caps>
                    <q-icon name="file_copy" />
                    <q-tooltip
                        transition-show="scale"
                        transition-hide="jump-down"
                        >Copy passcode</q-tooltip
                    >
                    Copy
                </q-btn>
            </q-btn-group>
        </q-form>
    </q-page>
</template>

<style scoped lang="scss">
#passcode {
    #form {
        .q-toolbar__title {
            padding: 15px;
        }
    }
}
</style>

<script setup lang="ts">
import { onMounted, ref } from 'vue';
import { usePasscodeStore } from '../stores';
import {
    copyInput,
    getPasscode,
    MAXIMUM_DIGIT,
    MAXIMUM_PASSCODE_LENGTH,
    MINIMUM_DIGIT,
    MINIMUM_PASSCODE_LENGTH
} from '../utils';

const store = usePasscodeStore();
const firstDigit = ref<number>();
const length = ref<number>();
const passcode = ref<number>();

onMounted(() => {
    firstDigit.value = store.firstDigit;
    length.value = store.length;
    passcode.value = store.passcode;
});
const onChange = () => {
    store.setFirstDigit(firstDigit.value);
    store.setLength(length.value);
    store.setPasscode(passcode.value);
};
const onCopy = () => {
    if (passcode.value) {
        const message = `Passcode ${passcode.value} copied to the clipboard!`;
        copyInput(passcode.value, message);
    }
};
const onReset = () => {
    delete firstDigit.value;
    delete length.value;
    delete passcode.value;
    store.resetFirstDigit();
    store.resetLength();
    store.resetPasscode();
};
const onSubmit = () => {
    store.setFirstDigit(firstDigit.value);
    store.setLength(length.value);
    passcode.value = getPasscode(length.value, firstDigit.value);
    store.setPasscode(passcode.value);
};
</script>
