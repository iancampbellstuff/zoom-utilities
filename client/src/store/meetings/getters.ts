// externals
import { GetterTree } from 'vuex';

// types
import { IMeetingsState } from './state';
import { IStore } from '../index';

const getters: GetterTree<IMeetingsState, IStore> = {
    getCurrentUserId(state: IMeetingsState) {
        return state.currentUserId;
    },
    getMeetings(state: IMeetingsState) {
        return state.meetings;
    },
    getSearch(state: IMeetingsState) {
        return state.search;
    },
    getUserIds(state: IMeetingsState) {
        return state.userIds;
    }
};

export default getters;
