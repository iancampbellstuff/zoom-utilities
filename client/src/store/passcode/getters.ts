import { GetterTree } from 'vuex';
import { IStore } from '../index';
import { IPasscodeState } from './state';

const getters: GetterTree<IPasscodeState, IStore> = {
    getFirstDigit(state: IPasscodeState) {
        return state.firstDigit;
    },
    getLength(state: IPasscodeState) {
        return state.length;
    },
    getPasscode(state: IPasscodeState) {
        return state.passcode;
    }
};

export default getters;
