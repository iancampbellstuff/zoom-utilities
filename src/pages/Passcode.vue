<template>
    <q-page id="passcode" class="row items-center justify-evenly">
        <q-form
            id="form"
            class="q-gutter-sm"
            @change="onChange"
            @submit="onSubmit"
            @reset="onReset"
            greedy
        >
            <q-toolbar-title class="text-center"
                >Generate a Passcode</q-toolbar-title
            >
            <q-input
                type="number"
                :min="minimumPasscodeLength"
                :max="maximumPasscodeLength"
                v-model="length"
                label="Length"
                hint="(optional)"
                outlined
                standout
                dense
            />
            <q-input
                type="number"
                :min="minimumDigit"
                :max="maximumDigit"
                v-model="firstDigit"
                label="First Digit"
                hint="(optional)"
                outlined
                standout
                dense
            />
            <q-btn-group spread>
                <q-btn label="Submit" type="submit" color="primary" no-caps />
                <q-btn
                    label="Reset"
                    type="reset"
                    color="primary"
                    class="q-ml-sm"
                    no-caps
                    flat
                />
            </q-btn-group>
            <q-btn-group spread>
                <q-input
                    v-model="passcode"
                    label="Passcode"
                    ref="passcode"
                    outlined
                    standout
                    dense
                    filled
                    readonly
                />
                <q-btn color="secondary" @click="onCopy" no-caps>
                    <q-icon name="file_copy" />
                    <q-tooltip
                        transition-show="scale"
                        transition-hide="jump-down"
                        >Copy passcode</q-tooltip
                    >
                    Copy
                </q-btn>
            </q-btn-group>
        </q-form>
    </q-page>
</template>

<style lang="scss" scoped>
#passcode {
    #form {
    }
}
</style>

<script lang="ts">
import Vue from 'vue';
import {
    copyInput,
    getPasscode,
    MAXIMUM_DIGIT,
    MAXIMUM_PASSCODE_LENGTH,
    MINIMUM_DIGIT,
    MINIMUM_PASSCODE_LENGTH
} from '../utils';
interface IData {
    firstDigit?: number;
    length?: number;
    maximumDigit: number;
    maximumPasscodeLength: number;
    minimumDigit: number;
    minimumPasscodeLength: number;
    passcode?: number;
}
export default Vue.extend({
    name: 'Passcode',
    data(): IData {
        return {
            firstDigit: undefined,
            length: undefined,
            maximumDigit: MAXIMUM_DIGIT,
            maximumPasscodeLength: MAXIMUM_PASSCODE_LENGTH,
            minimumDigit: MINIMUM_DIGIT,
            minimumPasscodeLength: MINIMUM_PASSCODE_LENGTH,
            passcode: undefined
        };
    },
    methods: {
        onChange() {
            this.$store.commit('passcodeModule/setLength', this.length);
            this.$store.commit('passcodeModule/setFirstDigit', this.firstDigit);
            this.$store.commit('passcodeModule/setPasscode', this.passcode);
        },
        onCopy() {
            const ref = this.$refs.passcode as HTMLInputElement;
            if (ref && this.passcode) {
                const message = `Passcode ${this.passcode} copied to the clipboard!`;
                copyInput(ref, message);
            }
        },
        onReset() {
            this.length = undefined;
            this.firstDigit = undefined;
            this.passcode = undefined;
            this.$store.commit('passcodeModule/resetLength');
            this.$store.commit('passcodeModule/resetFirstDigit');
            this.$store.commit('passcodeModule/resetPasscode');
        },
        onSubmit() {
            this.passcode = getPasscode(this.length, this.firstDigit);
            this.$store.commit('passcodeModule/setLength', this.length);
            this.$store.commit('passcodeModule/setFirstDigit', this.firstDigit);
            this.$store.commit('passcodeModule/setPasscode', this.passcode);
        }
    },
    mounted: function() {
        this.length = this.$store.getters['passcodeModule/getLength'];
        this.firstDigit = this.$store.getters['passcodeModule/getFirstDigit'];
        this.passcode = this.$store.getters['passcodeModule/getPasscode'];
    }
});
</script>
