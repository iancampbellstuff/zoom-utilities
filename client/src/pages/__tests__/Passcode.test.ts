import { createLocalVue, mount, Wrapper } from '@vue/test-utils';
import Vuex from 'vuex';
// code under test
import Passcode from '../Passcode.vue';
import getters from '../../store/passcode/getters';

const localVue = createLocalVue();
localVue.use(Vuex);

describe('Passcode', () => {
    let store;
    let wrapper: Wrapper<any>;
    beforeEach(() => {
        store = new Vuex.Store({
            modules: {
                passcodeModule: {
                    getters,
                    state: {}
                }
            }
        });
        wrapper = mount(Passcode, {
            localVue,
            store,
            stubs: [
                'q-btn-group',
                'q-btn',
                'q-form',
                'q-icon',
                'q-input',
                'q-page',
                'q-toolbar-title',
                'q-tooltip',
                'router-view'
            ]
        });
    });
    test('is a Vue instance', () => {
        expect(wrapper.vm).toBeTruthy();
    });
    it('should render correctly', () => {
        expect(wrapper.element).toMatchSnapshot();
    });
});
