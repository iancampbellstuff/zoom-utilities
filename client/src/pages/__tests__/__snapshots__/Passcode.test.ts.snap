// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Passcode should render correctly 1`] = `<!---->`;
